digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/RoundaboutProcessor.scala

```diff
@@ -0,0 +1,85 @@
+package fi.liikennevirasto.digiroad2.util
+
+import fi.liikennevirasto.digiroad2.asset.{Roundabout, State, TrafficDirection}
+import fi.liikennevirasto.digiroad2.linearasset.{RoadLink, RoadLinkLike}
+import fi.liikennevirasto.digiroad2.{GeometryUtils, Point}
+
+case class RoundaboutChangeSet(trafficDirectionChanges: Seq[RoadLink] = Seq.empty)
+
+object RoundaboutProcessor {
+
+  private def findRoundaboutRecursive(firstLink: RoadLink, nextLink: RoadLink, roadLinks: Seq[RoadLink], acc: Seq[RoadLink] = Seq.empty, withIncomplete: Boolean = true): (Seq[RoadLink], Seq[RoadLink]) = {
+    val (adjacents, rest) = roadLinks.partition(r => GeometryUtils.areAdjacent(r.geometry, nextLink.geometry))
+    adjacents match{
+      case Seq() if acc.size < 3 && !withIncomplete => {
+        println(s"findRoundaboutRecursive: ")
```

Comment:
We should use the logger. And it the message should be incomplete roundabout found.